### PR TITLE
Update Dockerfile packages

### PR DESCRIPTION
**Pull request by JasmondL**
#### Motivation
Fix vulnerabilities in the ModelMesh Controller image.

#### Modifications
1. Remove dependencies to any packages in the runtime. 
2. Updated Go version to the latest version.

#### Result 
The feature is functionally working, I have tested it on my own cluster.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -34,7 +34,7 @@ RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o manager
 ###############################################################################
 # Stage 2: Copy build assets to create the smallest final runtime image
 ###############################################################################
-FROM registry.access.redhat.com/ubi8/ubi-minimal:8.4 AS runtime
+FROM scratch AS runtime
```
Comment (ckadner) [inline Dockerfile:37]: I think the most suitable UBI image equivalent to `scratch` here would be `ubi-micro`, i.e. `registry.access.redhat.com/ubi8/ubi-micro:8.7` but I have not used it before. 

We should make sure that a minimal set of tools/CLIs are available to log on to a shell on the pod, `cat` a config or log file, etc.

What do you think @njhill

File: Dockerfile.develop [hunk 2]
```diff
@@ -15,9 +15,9 @@
 ###############################################################################
 # Create the develop, test, and build environment
 ###############################################################################
-FROM registry.access.redhat.com/ubi8/ubi-minimal:8.4
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.7
```
Comment (ckadner) [inline Dockerfile.develop:18]: Should we go bigger here and jump to UBI 9.1.0. I had tried that a week ago and it seemed to work fine.

**Review by ckadner — CHANGES_REQUESTED** [review]
Thanks @JasmondL -- I have a few questions, comments :-)

File: go.mod [hunk 3]
```diff
@@ -1,70 +1,68 @@
...
+go 1.19
 
 require (
 	github.com/dereklstinson/cifar v0.0.0-20200421171932-5722a3b6a0c7
 	github.com/go-logr/logr v1.2.3
 	github.com/golang/protobuf v1.5.2
-	github.com/google/go-cmp v0.5.8
-	github.com/kserve/kserve v0.9.1-0.20221013012311-335dfbcc461a
+	github.com/google/go-cmp v0.5.9
+	github.com/kserve/kserve v0.9.0
```
Comment (ckadner) [inline go.mod:10]: can you change the `kserve` version to [`v0.10.0-rc0`](https://github.com/kserve/kserve/releases/tag/v0.10.0-rc0)

```shell
$ go get github.com/kserve/kserve@v0.10.0-rc0

go: downloading github.com/kserve/kserve v0.10.0-rc0
go: upgraded github.com/kserve/kserve v0.9.1-0.20221013012311-335dfbcc461a => v0.10.0-rc0

```

File: Dockerfile.develop (hunk 2 above)
Comment (ckadner) [reply Dockerfile.develop:18]: @njhill suggested to stick with UBI `8.7` so this is good :-)

File: Dockerfile.develop [hunk 4]
```diff
@@ -15,9 +15,9 @@
...
 
-ARG GOLANG_VERSION=1.18.7
+ARG GOLANG_VERSION=1.19.3
```
Comment (ckadner) [inline Dockerfile.develop:20]: we should leave the Go version at 1.18 for the time being, but we can update to the latest service version `1.18.9`

https://go.dev/dl/go1.18.9.linux-amd64.tar.gz

File: go.mod [hunk 5]
```diff
@@ -1,70 +1,68 @@
 module github.com/kserve/modelmesh-serving
 
-go 1.18
+go 1.19
```
Comment (ckadner) [inline go.mod:3]: keep at `1.18`

**Review by ckadner — CHANGES_REQUESTED** [review]
a few more comments after chatting with @njhill

File: go.mod (hunk 5 above)
Comment (JasmondL) [reply go.mod:3]: Will apply on the next commit.

File: Dockerfile.develop (hunk 4 above)
Comment (JasmondL) [reply Dockerfile.develop:20]: Will apply on the next commit.

File: go.mod [hunk 6]
```diff
@@ -124,7 +124,11 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
 )
 
-replace go.uber.org/atomic => github.com/uber-go/atomic v1.9.0
+replace (
+  github.com/kserve/kserve => github.com/kserve/kserve v0.10.0-rc0
```
Comment (ckadner) [inline go.mod:128]: Thanks for your new changes @JasmondL

To fix the `missing go.sum entry` errors, please run `go mod tidy` and add/commit the updated `go.sum` file

**Review by ckadner — CHANGES_REQUESTED** [review]

File: go.mod (hunk 6 above)
Comment (JasmondL) [reply go.mod]: Done

**Review by ckadner — APPROVED** [review]
/lgtm

**Comment by ckadner** [comment]
Thanks @JasmondL -- this looks good now!

/cc @njhill

**Review by njhill — APPROVED** [review]
Thanks @JasmondL @ckadner

**Pull request merged**